### PR TITLE
added contact developer and demo for rating in settings

### PR DESCRIPTION
**Pull request by tasneem22**

### CODE REVIEW

File: lib/screens/settings_screen.dart [hunk 1]
```diff
@@ -51,7 +54,46 @@ class _SettingsScreenState extends State<SettingsScreen> {
...
+            ),
+            ListTile(
+              title: const Text("Rate us"),
+              trailing: RatingBar.builder(
+                initialRating: 0,
+                minRating: 1,
+                direction: Axis.horizontal,
+                allowHalfRating: true,
+                itemCount: 5,
+                itemPadding: EdgeInsets.symmetric(horizontal: 4.0),
```
Comment (aabounegm) [inline lib/screens/settings_screen.dart:86]: "Prefer const with constant constructors" - dart analyzer.
Please make sure that Android Studio shows you Dart Analyzer's messages

File: lib/screens/settings_screen.dart [hunk 2]
```diff
@@ -51,7 +54,46 @@ class _SettingsScreenState extends State<SettingsScreen> {
...
+                onPressed: () async =>
+                    await launch("https://t.me/taasneemtoolba"),
+                child: const Text('Open Telegram'),
+                color: Colors.black,
+
+              ),
+            ),
+            ListTile(
+              title: const Text("Rate us"),
+              trailing: RatingBar.builder(
```
Comment (aabounegm) [inline lib/screens/settings_screen.dart:80]: Do we really need this `RatingBar` package? It's not like the rating actually goes to the Play/App Store. I think it should just be a button that redirects to the relevant store.

File: lib/screens/settings_screen.dart [hunk 3]
```diff
@@ -51,7 +54,46 @@ class _SettingsScreenState extends State<SettingsScreen> {
...
+                itemBuilder: (context, _) => const Icon(
+                  Icons.star,
+                ),
+                onRatingUpdate: (rating) {
+                  rating = rating;
+                },
+              ),
+            ),
+          ],
+           )
```
Comment (aabounegm) [inline lib/screens/settings_screen.dart:96]: There is an extra parenthesis left over from the merge conflict. This is what's preventing it from building

**Review by aabounegm — CHANGES_REQUESTED** [review]
Please make sure that the CI always passes. In this case, it was because of a syntax error from the merge conflict.
Also, I didn't check any changes under the `ios` folder as I don't understand what goes on there 😁

File: lib/screens/settings_screen.dart [hunk 4]
```diff
@@ -113,4 +126,11 @@ class _SettingsScreenState extends State<SettingsScreen> {
       await FirebaseAuth.instance.signInWithCredential(cred);
     }
   }
+  Future<void> _requestReview() async{
+    if (await _inAppReview.isAvailable()) {
+    return await _inAppReview.requestReview();
+    }
+  }
+  // => _inAppReview.requestReview();
```
Comment (aabounegm) [inline lib/screens/settings_screen.dart:134]: What is this comment?

File: lib/screens/settings_screen.dart [hunk 5]
```diff
@@ -55,45 +82,31 @@ class _SettingsScreenState extends State<SettingsScreen> {
...
-             const Center(
-                  child: Text(
-                "contact developer",
-                style: TextStyle(fontSize: 23),
-              )),
+            const Center(
+                child: Text(
+              "contact developer",
+              style: TextStyle(fontSize: 23),
+            )),
```
Comment (aabounegm) [inline lib/screens/settings_screen.dart:89]: Don't forget trailing commas for better formatting

File: lib/screens/settings_screen.dart [hunk 6]
```diff
@@ -15,7 +18,31 @@ class SettingsScreen extends StatefulWidget {
...
+            // more information.
+            _availability = isAvailable && !Platform.isAndroid
+                ? Availability.available
+                : Availability.unavailable;
+          });
+        } catch (e) {
+          setState(() => _availability = Availability.unavailable);
+        }
+      });
+    }
```
Comment (aabounegm) [inline lib/screens/settings_screen.dart:44]: What are you using this `_availability` for?

File: lib/screens/settings_screen.dart [hunk 7]
```diff
@@ -15,7 +18,31 @@ class SettingsScreen extends StatefulWidget {
 
 class _SettingsScreenState extends State<SettingsScreen> {
   User? get user => FirebaseAuth.instance.currentUser;
-  double rating = 0;
+  final InAppReview _inAppReview = InAppReview.instance;
+  Availability _availability = Availability.loading;
+
+  @override
+    void initState() {
```
Comment (aabounegm) [inline lib/screens/settings_screen.dart:25]: What is this extra level of indentation? Are you sure you are using the dart formatter?

File: lib/constants.dart [hunk 8]
```diff
@@ -0,0 +1,2 @@
+const kAppStoreId = '';
+const kPlayStoreId = '';
```
Comment (aabounegm) [inline lib/constants.dart:2]: I don't see these constants being used anywhere

**Review by aabounegm — CHANGES_REQUESTED** [review]
When you request review, please make sure that all previous comments are addressed, that there are no merge conflicts, and that the CI passes.
In this case, the CI did not run because there are many merge conflicts. Please fix them first, and make sure you set up autoformatting with dartfmt.

File: lib/screens/settings_screen.dart [hunk 9]
```diff
@@ -19,29 +15,11 @@ class SettingsScreen extends StatefulWidget {
...
-                : Availability.unavailable;
-          });
-        } catch (e) {
-          setState(() => _availability = Availability.unavailable);
-        }
-      });
-    }
+  void initState() {
+    super.initState();
+  }
```
Comment (aabounegm) [inline lib/screens/settings_screen.dart:22]: This override does nothing. You can remove it and `super.initState` will be implicitly called  anyways

**Review by aabounegm — APPROVED** [review]

**Comment by alkaitagi** [comment]
@aabounegm must we close this?

**Comment by aabounegm** [comment]
Yes I think we should. It would take more effort to bring it up to date than to reimplement the added feature

**Pull request closed without merging**